### PR TITLE
Plateau degeu

### PR DESCRIPTION
**Pull request by QuentinRimet**
- création d'un plateau de jeu et modélisation de celui-ci en carré
- chaque thème possède sa propre couleur
- ajout de carré de couleur pour les joueurs
- 4 joueurs par plateau
- utilisation de twig
- route définie dans l'index
- suivant le modèle mvc

### CODE REVIEW

**Comment by ElGitariste** [comment]
Ok ça à l'air bon !

File: index.php [hunk 1]
```diff
@@ -40,6 +40,13 @@
 	$acc->affichageAcc();
 })->setName('Accueil');
 
+
+$app->get('/plateau', function() {
+	$acc = new jeuControlleur();
```
Comment (floo51) [inline index.php]: Naming should be in english.

File: index.php [hunk 2]
```diff
@@ -40,6 +40,13 @@
...
+	$acc->initJeu();
+	$acc->affichageJeu();
```
Comment (floo51) [inline index.php]: A method should start with a verb, not a noun. It's an action, not an object.

File: src/Jeu.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class jeu {
+    private $plateau= array();
+
+    public function __construct(){
+        $this->plateau = new Plateau();
+        $this->plateau->creaPlateau();
+    }
+
+    public function getPlateau(){
```
Comment (floo51) [inline src/Jeu.php]: Is there a point having a getter instead of making your `$plateau` property public?

File: src/Plateau.php [hunk 4]
```diff
@@ -1,137 +1,92 @@
...
 
-for($i=1;$i<=$taille_tab;$i++){
-    for($j=1;$j<=$taille_tab;$j++){
-        if($i-0.5==$taille_tab/2 && $j-0.5==$taille_tab/2 ){
-            $tableau[$i][$j]="depart";
-        }
-        else
-        if($i==1 || ($i-0.5==$taille_tab/2 && $j<$taille_tab/2)){
+    public function creaPlateau(){
+    for($i=1;$i<=$this->taille_tab;$i++){
```
Comment (floo51) [inline src/Plateau.php]: This is hardly readable, some clues:
- Can you write this code in another way? Maybe just initialize the table without any for loops (the table doesn't change size right?)
- Can you try to externalize some of the code to named functions?
- Can you take more care with variable naming? `$i` and `$j` usually can be replaced with things like `$column` and `$row`

File: src/Plateau.php [hunk 5]
```diff
@@ -1,137 +1,92 @@
...
+        for($j=1;$j<=$this->taille_tab;$j++){
+            if($i-0.5==$this->taille_tab/2 && $j-0.5==$this->taille_tab/2 ){
+                $this->tableau[$i][$j]=["theme"=>"depart","player"=>array()];
+            }
+            else
+            if($i==1 || ($i-0.5==$this->taille_tab/2 && $j<$this->taille_tab/2)){
+                switch($j%6){
+                    case 0:$this->tableau[$i][$j]=["theme"=>"geo","player"=>array()];
```
Comment (floo51) [inline src/Plateau.php]: `"geo", "diver", ...` are called magic strings and are hell for maintainability. Use constants instead?

File: src/Plateau.php [hunk 6]
```diff
@@ -1,137 +1,92 @@
...
+            $this->tableau[$i][$j]=["theme"=>null,"player"=>array()];
+        }
     }
+    
+    $this->tableau[7][7]['player']=["player1","player2","player3","player4"];
     }
-        else
-         $tableau[$i][$j]="";
+
+    function getTableau(){
```
Comment (floo51) [inline src/Plateau.php]: Is there a point having a getter instead of a public property?

File: src/Plateau.php [hunk 7]
```diff
@@ -1,137 +1,92 @@
...
-}
-echo "<table>";
-for($i=1;$i<=$taille_tab;$i++){
-    echo "<tr>";
-    for($j=1;$j<=$taille_tab;$j++){
-        echo "<td class={$tableau[$i][$j]}>";
-        
-        echo "</td>";
+    function getTaille(){
+        return $this->taille_tab;
```
Comment (floo51) [inline src/Plateau.php]: Does the game board size actually change? If not, why use a variable?

File: src/vues/JeuVue.php [hunk 8]
```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace trivial\vues;
+
+use trivial\vues\GlobaleVue;
+
+class JeuVue {
+
+    public function renderPlateau($plateau){
```
Comment (floo51) [inline src/vues/JeuVue.php]: This is not OK. You can't have a templating engine AND generate html with string concatenation.

File: src/vues/JeuVue.php [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+            $html .= "</tr>";
+        }
+    
+        $html .= "</table>";
+    
+        return $html;
+        }
+
+	public function render($a) {
+        $ach = GlobaleVue::header();
```
Comment (floo51) [inline src/vues/JeuVue.php]: Be careful with variable naming, I have no idea what `$ach` and `$acf` refer to.

**Review by floo51 — CHANGES_REQUESTED** [review]

File: container.php [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+use trivial\controlleurs\GameController as GameController;
+
+$configuration = [
+    'settings' => [
+        'displayErrorDetails' => true,
+    ],
+];
+$container = new \Slim\Container($configuration);
+$container['GameController'] = function($c) {
+	$view = $c->get("view"); // retrieve the 'view' from the container
```
Comment (floo51) [inline container.php]: Not sure the comment is needed there, the code explains itself.

File: container.php [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+	return new GameController($view);
+};
+
+$container['view'] = function ($container) {
+    $view = new \Slim\Views\Twig('src/vues');
+
+    // Instantiate and add Slim specific extension
```
Comment (floo51) [inline container.php:20]: This one seems much more needed because I'm not able to understand the 2 lines below 👍

File: src/Board.php [hunk 12]
```diff
@@ -0,0 +1,94 @@
+<?php
+namespace trivial;
+
+ class Board {
+    public $grid = array();
+
+    public function newBoard(){
+    $hist="hist";
```
Comment (floo51) [inline src/Board.php]: That's variables, not constants: https://stackoverflow.com/questions/2447791/define-vs-const

File: src/Board.php [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    $geo="geo";
+    $diver="diver";
+    $sport="sport";
+    $info="info";
+    $perso="perso"; 
+
+    $this->grid[1][1]=["theme"=>$hist,"player"=>array()];
```
Comment (floo51) [inline src/Board.php]: This is, to me, much more readable than all the loops and switches 👍

File: src/vues/GameView.html.twig [hunk 14]
```diff
@@ -0,0 +1,21 @@
+{% include 'GlobalHeaderView.html.twig' %}
+<table>
```
Comment (floo51) [inline src/vues/GameView.html.twig]: Not mandatory for this review, you might want to use more responsive html tags. `div` with css grid by instance: https://css-tricks.com/snippets/css/complete-guide-grid/

File: src/vues/GlobalHeaderView.html.twig [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+    <head>
+    <meta charset="utf-8">
+    <title>{% block titlre %}Trivial Navigo{% endblock %} </title>
+    <meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no">
+    {% block stylesheets %}
+        <link rel="stylesheet" href="web/css/board.css"/>
+    {% endblock %}
+    </head>
```
Comment (floo51) [inline src/vues/GlobalHeaderView.html.twig:8]: Seems like there are missing closing tags?

**Review by floo51 — CHANGES_REQUESTED** [review]

**Review by floo51 — APPROVED** [review]

**Pull request merged**